web/src/engine/websites/MechaComic.ts

```diff
@@ -0,0 +1,95 @@
+import { Tags } from '../Tags';
+import icon from './MechaComic.webp';
+import { Chapter, DecoratableMangaScraper, Page, type Manga } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { Fetch, FetchCSS, FetchJSON } from '../platform/FetchProvider';
+import type { Priority } from '../taskpool/DeferredTask';
+
+type ContentData = {
+    images: {
+        [id: string]: {
+            src: string,
+            format: string
+        }[]
+    }
+}
+
+@Common.MangaCSS(/^{origin}\/books\/\d+$/, 'div.p-bookInfo_title h1')
+@Common.MangasMultiPageCSS('/free/list?page={page}', 'div.p-book_detail dt.p-book_title a')
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('mechacomic', 'MechaComic', 'https://mechacomic.jp', Tags.Media.Manga, Tags.Language.Japanese, Tags.Source.Official);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const chapterList = [];
+        for (let page = 1, run = true; run; page++) {
+            const chapters = await this.GetChaptersFromPage(manga, page);
+            chapters.length > 0 ? chapterList.push(...chapters) : run = false;
+        }
+        return chapterList;
+    }
+
+    private async GetChaptersFromPage(manga: Manga, page: number): Promise<Chapter[]> {
+        const data = await FetchCSS(new Request(new URL(`${manga.Identifier}?page=${page}`, this.URI)), 'li.p-chapterList_item div.p-chapterInfo-comic');
+        return data.map(chapter => {
+            const title = [
+                Common.ElementLabelExtractor('span').call(this, chapter.querySelector('.p-chapterList_no')).trim(),
+                chapter.querySelector('.p-chapterList_name').textContent.trim()]
+                .join(' ').trim();
+            const link = chapter.querySelector<HTMLAnchorElement>('a');
+            return new Chapter(this, manga, link.pathname, title);
+        });
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const chapterUrl = new URL(chapter.Identifier, this.URI);
+        const response = await Fetch(new Request(chapterUrl));
+        const url = response.redirected ? new URL(response.url) : chapterUrl;
```

Comment:
This check seems redundant, isn't `response.url` always the correct URL that should be used?

Reply:
Yes but if no redirection occurs (who knows with those websites) response.url will undefined.
This way is safer, and its a simple check.